x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,138 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+	denom, err := k.iterateAndUpdateFundsDistribution(ctx)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+	// get allocated continuous fund
+	fundsAllocated, err := k.getAndResetDistributedFunds(ctx, recipient)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while getting distributed funds: %w", err)
+	}
+
+	recipientBal := k.bankKeeper.GetAllBalances(ctx, recipient)
+	recipientAmount := recipientBal.AmountOf(denom)
+	totalRecipientBal := recipientAmount.Add(fundsAllocated)
+	// check if the recipient account balance exceeds maxDistributedCapital after distribution
+	var withdrawnAmount sdk.Coin
+	if totalRecipientBal.LT(cf.MaxDistributedCapital) {
+		withdrawnAmount = sdk.NewCoin(denom, fundsAllocated)
+		// Distribute funds to the recipient from pool module account
+		err := k.DistributeFromCommunityPool(ctx, sdk.NewCoins(withdrawnAmount), recipient)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+
+		// decrement fundsAllocated from ToDistribute
+		toDistribute, err := k.ToDistribute.Get(ctx)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+		err = k.ToDistribute.Set(ctx, toDistribute.Sub(fundsAllocated))
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+	}
+
+	return withdrawnAmount, nil
+}
+
+func (k Keeper) SetToDistribute(ctx context.Context, amount sdk.Coins) error {
```

Comment:
Do we really need this given that `ToDistribute` is public on the keeper?

Reply:
I think using mock methods would allow us to not introduce protocolpool module dependencies in keeper. 
We could directly use `ToDistribute` only if use protocolpool keeper is module dependent.
Please correct me if I'm wrong.